### PR TITLE
Angular init + authorization

### PR DESCRIPTION
**Pull request by ozhurova**
1. Angular init
2. Added Angular Material
3. Created main page, error page
4. Added api service, auth service, user service
5. Added auth guard
image
image
image
image
image

### CODE REVIEW

File: client/src/app/core/components/login-modal/login-modal.component.html [hunk 1]
```diff
@@ -0,0 +1,59 @@
+<app-modal-window [title]="title">
+
+  <form *ngIf="isLogin" class="login-wrapper" [formGroup]="loginForm">
```
Comment (Morphine-medicine) [inline client/src/app/core/components/login-modal/login-modal.component.html:3]: looks good, but I'll recommend you to change structure to 
```
<ng-container *ngIf="isLogin; else registerTemplate">
<form class="login-wrapper" [formGroup]="loginForm">
....
</form>
</ng-container>

<ng-template #registerTemplate>
<form class="login-wrapper" [formGroup]="registerForm">
...
</form>
</ng-template>
```
OR you can create form, looking on isLogin, and you will have 1 template for both fuctions, and for Name formControl you can use ngIf

File: client/src/app/core/components/login-modal/login-modal.component.html [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+    </mat-form-field>
+
+    <mat-form-field class="login-field" appearance="fill">
+      <mat-label>Enter your password</mat-label>
+      <input name="password" type="password" matInput formControlName="password" required>
+      <mat-error *ngIf="registerForm.controls['password'].hasError('required')" class="input-error">Error</mat-error>
+    </mat-form-field>
+
+    <div class="btn-container">
+      <button mat-raised-button color="primary" class="mat-btn-primary" (click)="cancel()">
```
Comment (Morphine-medicine) [inline client/src/app/core/components/login-modal/login-modal.component.html:48]: no need to use ngIf because it's inside the same ngIf

**Review by Morphine-medicine — CHANGES_REQUESTED** [review]
Looks really great! Take a look on comments, and after fixing it, let me know, even if it will be done after deadline, I'll approve that HW 5 is done, great job!

**Comment by Morphine-medicine** [comment]
And also, I'll recommend toi add some paddings for header, mb 16px it will look better)

File: client/src/app/core/components/login-modal/login-modal.component.html (hunk 1 above)
Comment (ozhurova) [reply client/src/app/core/components/login-modal/login-modal.component.html:3]: if I make one generic template and one form, but how do I determine which user I received in order to make the correct api request?

Comment (ozhurova) [reply client/src/app/core/components/login-modal/login-modal.component.html]: I changed to <ng-container> and <ng-template>

File: client/src/app/core/components/login-modal/login-modal.component.html (hunk 2 above)
Comment (ozhurova) [reply client/src/app/core/components/login-modal/login-modal.component.html]: I fixed it on a new branch, but now I've fixed it here

File: client/src/app/core/components/header/header.component.html [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+      <button *ngIf="!currentUserName" mat-raised-button color="primary" class="btn-login" (click)="openModal()">
+        Log in
+      </button>
+      <button *ngIf="currentUserName" mat-raised-button color="primary" class="btn-login" (click)="logout()">
+        Log out
+      </button>
+    </div>
+  </div>
+</header>
+<app-login-modal *ngIf="isVisible" (modal)="closeModal()" [isLogin]="isLogin" [title] = title></app-login-modal>
```
[the remark this answers is not in the document]
Comment (ozhurova) [reply client/src/app/core/components/header/header.component.html]: fixed it

**Pull request merged**